### PR TITLE
DAOS-6452 test: Use JSON for dmg pool query and update tests

### PR DESCRIPTION
**Pull request by shimizukko**
Use self._get_json_result in pool_query()

Update tests that use pool_query()

### CODE REVIEW

File: src/tests/ftest/util/dmg_utils.py [hunk 1]
```diff
@@ -437,115 +437,45 @@ def pool_query(self, pool):
...
+        #             "free": 31999950848,
+        #             "min": 3999993856,
+        #             "max": 3999993856,
+        #             "mean": 3999993856
+        #         }
+        #     },
+        #     "error": null,
+        #     "status": 0
+        # }
+        return self._get_json_result(("pool", "query"), pool=pool)
```
Comment (saurabhtandan) [inline src/tests/ftest/util/dmg_utils.py:478]: Can we provide a provision where we can use both json and non json output if somebody calls pool_query().
Supplying a simple flag like pool_query(pool, json=True/False) should enable or disable json or non json output.

**Review by saurabhtandan — CHANGES_REQUESTED** [review]
As there are changes in util files which could affect wide range of tests, I would suggest running complete PR tests along with pool query tests.

**Review by shimizukko — COMMENTED** [review]
I think it's a good idea to add an argument such as pool_query(json=True). In my other PR, I want to call dmg pool create without --json to see the warning message, but we only use _get_json_result. @phender What do you think?

**Review by dinghwah — COMMENTED** [review]
Since there are updates in dmg_pool_utils and test_util_pool, could you please add more tests in pool and container to make sure backward compatibilities.

**Comment by shimizukko** [comment]
> Since there are updates in dmg_pool_utils and test_util_pool, could you please add more tests in pool and container to make sure backward compatibilities.

Not sure what you mean. dmg_pool_query_test.py covers most of the output fields and the OSA tests cover the rebuild fields. All the PR tests were executed.

File: src/tests/ftest/util/dmg_utils.py [hunk 2]
```diff
@@ -427,125 +427,61 @@ def pool_create(self, scm_size, uid=None, gid=None, nvme_size=None,
 
         return data
 
-    def pool_query(self, pool):
+    def pool_query(self, pool, use_json=True):
```
Comment (phender) [inline src/tests/ftest/util/dmg_utils.py:430]: I would not recommend this approach.  We should eventually be converting all of these `DmgCommand` helper methods to use json output, so why not fully convert this one?

File: src/tests/ftest/util/dmg_utils.py [hunk 3]
```diff
@@ -427,125 +427,61 @@ def pool_create(self, scm_size, uid=None, gid=None, nvme_size=None,
...
+        #             "mean": 3999993856
+        #         }
+        #     },
+        #     "error": null,
+        #     "status": 0
+        # }
+        if use_json:
+            return self._get_json_result(("pool", "query"), pool=pool)
+
+        return self._get_result(("pool", "query"), pool=pool)
```
Comment (phender) [inline src/tests/ftest/util/dmg_utils.py:484]: This really should be just:
```suggestion
        return self._get_json_result(("pool", "query"), pool=pool)
```

Any callers of this method will just have need to implement this conversion:
```
data["uuid"]               -> result["response"]["uuid"]
data["ntarget"]            -> result["response"]["total_targets"]
data["disabled"]           -> result["response"]["disabled_targets"]
data["leader"]             -> result["response"]["leader"]
data["version"]            -> result["response"]["version"]
data["target_count"]       -> result["response"]["uuid"]
data["scm"]["total"]       -> result["response"]["scm"]["total"]
data["scm"]["free"]        -> result["response"]["scm"]["free"]
data["scm"]["free_min"]    -> result["response"]["scm"]["min"]
data["scm"]["free_max"]    -> result["response"]["scm"]["max"]
data["scm"]["free_mean"]   -> result["response"]["scm"]["mean"]
data["nvme"]["total"]      -> result["response"]["nvme"]["total"]
data["nvme"]["free"]       -> result["response"]["nvme"]["free"]
data["nvme"]["free_min"]   -> result["response"]["nvme"]["min"]
data["nvme"]["free_max"]   -> result["response"]["nvme"]["max"]
data["nvme"]["free_mean"]  -> result["response"]["nvme"]["mean"]
data["rebuild"]["status"]  -> result["response"]["rebuild"]["state"]
data["rebuild"]["status2"] -> result["response"]["rebuild"]["objects"]
data["rebuild"]["status3"] -> result["response"]["rebuild"]["records"]
```

**Review by phender — CHANGES_REQUESTED** [review]

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Review by dinghwah — APPROVED** [review]
LGTM.

**Comment by shimizukko** [comment]
All 3 test failures are DAOS-7016.

File: src/tests/ftest/util/dmg_utils.py (hunk 2 above)
Comment (shimizukko) [reply src/tests/ftest/util/dmg_utils.py:430]: Thanks for the review. I'll remove this argument in my next PR.

**Pull request merged**